ssl/ssl_quic.c

```diff
@@ -178,70 +178,89 @@ int SSL_provide_quic_data(SSL *ssl, OSSL_ENCRYPTION_LEVEL level,
     if (len == 0)
         return 1;
 
-    if (sc->quic_buf == NULL) {
-        BUF_MEM *buf;
-        if ((buf = BUF_MEM_new()) == NULL) {
-            ERR_raise(ERR_LIB_SSL, ERR_R_INTERNAL_ERROR);
-            return 0;
-        }
-        if (!BUF_MEM_grow(buf, SSL3_RT_MAX_PLAIN_LENGTH)) {
-            ERR_raise(ERR_LIB_SSL, ERR_R_INTERNAL_ERROR);
-            BUF_MEM_free(buf);
-            return 0;
-        }
-        sc->quic_buf = buf;
-        /* We preallocated storage, but there's still no *data*. */
-        sc->quic_buf->length = 0;
-        buf = NULL;
-    }
-
     /* A TLS message must not cross an encryption level boundary */
-    if (sc->quic_buf->length != sc->quic_next_record_start
-            && level != sc->quic_latest_level_received) {
+    fragment_length = ssl->quic_buf != NULL
+                        ? fragment_length = ssl->quic_buf->length : 0;
```

Comment:
It looks like there's an extra "fragment_length =" in there.

Reply:
oops.  fixup pushed!  (And I moved some variable declarations while I was there). Thank!